platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/type/ProjectTypeDef.java

```diff
@@ -118,25 +142,21 @@ protected void addAttributeDefinition(Attribute attr) {
         attributes.put(attr.getName(), attr);
     }
 
-    protected void addParent(AbstractProjectType parent) {
-        parents.add(parent);
-    }
-
-    private boolean recurseParents(ProjectType child, String parent) {
 
-        for (ProjectType p : child.getParents()) {
-            if (p.getId().equals(parent)) {
-                return true;
-            }
-            if (recurseParents(p, parent)) {
-                return true;
-            }
+    protected void addParent(String parentId) {
+        for(String pid : parents) {
+            if(pid.equals(parentId))
+                return;
         }
+        parents.add(parentId);
+    }
 
-        return false;
+    void addAncestor(String ancestor) {
+        this.ancestors.add(ancestor);
     }
 
-    @Override
+
+    //@Override
```

Comment:
it is not needed, remove comment and maybe the method as it's not overriding anymore and maybe nobody is using it